File: blog_app/blog_app/src/main/java/com/example/blog/service/CommentService.java
package com.example.blog.service;

import com.example.blog.entities.Comment;
import com.example.blog.utils.CommentDto;

import java.util.List;

public interface CommentService {

    CommentDto createComment(CommentDto commentDto, Integer postId);

    void deleteComment(Integer commentId);

    List<CommentDto> getCommentsByPost(Integer postId);
}

File: blog-app-main/src/services/comment-service.js
import { privateAxios, myAxios } from "./helper";

export const createComment = (comment, postId) => {
  return privateAxios.post(`/api/post/${postId}/comments`, comment)
    .then((response) => response.data);
};

export const getCommentsByPost = (postId) => {
  return myAxios.get(`/api/post/${postId}/comments`)
    .then((response) => response.data);
};

export const deleteComment = (commentId) => {
  return privateAxios.delete(`/api/comments/${commentId}`)
    .then((response) => response.data);
};